CMakeLists.txt

```diff
@@ -13,6 +13,11 @@ set (SOVERSION "10")
 
 include(GNUInstallDirs)
 include(GeneratePkgConfig)
+include(CMakeDependentOption)
+
+# Boost
+find_public_dependency(Boost REQUIRED COMPONENTS system)
+find_public_dependency(Boost REQUIRED COMPONENTS date_time)
```

Comment:
The `REQUIRED` keyword has to be removed.

Reply:
`date_time` is not an unconditional public dependency, but it is only when `deprecated-functions` is set (according to your code below).